### ISSUE TITLE
req.hostname does not include port number on some environments

### ISSUE THREAD

**Issue by alevito** [Financial-Times/s3o-middleware#54]
Changes introduced by this PR https://github.com/Financial-Times/s3o-middleware/pull/51/files are causing s3o to redirect to the original hostname without port.
Apparently I'm always redirected to the host without the port. e.g., If the original host is local.ft.com:3005, after going through s3o, my request is then redirected to local.ft.com

As a work around, on my machine (Fedora 29) I had to set up an nginx reverse proxy with the following configuration.
```
server {
  listen 80;
  listen [::]:80;
  server_name   local.ft.com;

  location / {
      proxy_pass http://local.ft.com:3005;
  }
}
```

**Comment by CourtneyLO** [comment]
I have had the same issue when trying to run [biz-ops-api](https://github.com/Financial-Times/biz-ops-api) locally.

**Comment by cebirmingham** [comment]
I am having the same issue, the port is getting removed after the redirect when running locally

**Comment by rowanmanning** [comment]
Thanks for reporting everyone 🙂

This should be fixed in v3.0.1, which you can upgrade to with

```npm install @financial-times/s3o-middleware@latest```

We can reopen this if you're still having issues.
